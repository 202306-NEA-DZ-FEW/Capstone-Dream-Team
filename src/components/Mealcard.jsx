import React from "react";

export default function Mealcard() {
    return (
        <div className='w-80 h-96 relative'>
            <div className='w-80 h-96 left-0 top-0 absolute bg-zinc-100 rounded-3xl' />
            <div className='w-72 h-80 left-[16.02px] top-[40.94px] absolute'>
                <div className='w-60 h-11 left-[40.78px] top-[124.90px] absolute'>
                    <div className='w-60 h-11 left-0 top-0 absolute bg-rose-100 rounded-2xl' />
                    <div className="w-40 h-3.5 left-[17.20px] top-[9.17px] absolute text-center text-black text-base font-normal font-['Roboto'] leading-relaxed">
                        Your name (optional)
                    </div>
                </div>
                <div className='w-60 h-11 left-[40.78px] top-[302.65px] absolute'>
                    <div className='w-60 h-11 left-0 top-0 absolute bg-rose-400 rounded-full' />
                    <div className="w-32 h-3.5 left-[57.88px] top-[10.42px] absolute text-center text-black text-xl font-bold font-['Roboto'] leading-relaxed">
                        Donate (12$)
                    </div>
                </div>
                <div className='w-60 h-20 left-[40.78px] top-[190.17px] absolute'>
                    <div className='w-60 h-20 left-0 top-0 absolute bg-rose-100 rounded-2xl shadow' />
                    <img
                        className='w-5 h-5 left-[208.19px] top-[59.37px] absolute rounded-2xl'
                        src='https://via.placeholder.com/19x20'
                    />
                    <div className="w-48 h-3 left-[17.20px] top-[10.89px] absolute text-center text-black text-base font-normal font-['Roboto'] leading-relaxed">
                        Say something nice (optional)
                    </div>
                </div>
                <div className='w-60 h-20 left-[40.78px] top-[32.08px] absolute'>
                    <div className='w-60 h-20 left-0 top-0 absolute bg-rose-100 rounded-2xl' />
                    <div className='w-48 h-10 left-[19.13px] top-[17.70px] absolute'>
                        <div className='w-7 h-8 left-[123.53px] top-[5.60px] absolute'>
                            <div className='w-7 h-8 left-0 top-0 absolute bg-red-200 rounded-full' />
                            <div className="w-4 h-3 left-[5.54px] top-[3.68px] absolute text-center text-black text-base font-bold font-['Roboto'] leading-relaxed">
                                2
                            </div>
                        </div>
                        <div className='w-7 h-8 left-[169.47px] top-[5.60px] absolute'>
                            <div className='w-7 h-8 left-0 top-0 absolute bg-red-200 rounded-full' />
                            <div className="w-4 h-3 left-[5.44px] top-[4.82px] absolute text-center text-black text-base font-bold font-['Roboto'] leading-relaxed">
                                5
                            </div>
                        </div>
                        <div className='w-7 h-8 left-[79.83px] top-[5.60px] absolute'>
                            <div className='w-7 h-8 left-0 top-0 absolute bg-rose-400 rounded-full' />
                            <div className="w-4 h-3 left-[5.44px] top-[3.82px] absolute text-center text-black text-base font-bold font-['Roboto'] leading-relaxed">
                                1
                            </div>
                        </div>
                        <div className="w-4 h-3 left-[52.97px] top-[8.42px] absolute text-center text-black text-xl font-bold font-['Roboto'] leading-relaxed">
                            X
                        </div>
                        <img
                            className='w-9 h-10 left-0 top-0 absolute rounded-2xl'
                            src='https://via.placeholder.com/37x39'
                        />
                    </div>
                </div>
                <div className="w-48 h-3.5 left-0 top-0 absolute text-center text-black text-xl font-bold font-['Roboto'] leading-relaxed">
                    BUY A MEAL{" "}
                </div>
            </div>
        </div>
    );
}
