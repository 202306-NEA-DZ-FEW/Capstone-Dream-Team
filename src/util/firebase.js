// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
