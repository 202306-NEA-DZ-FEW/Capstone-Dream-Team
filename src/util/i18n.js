import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
