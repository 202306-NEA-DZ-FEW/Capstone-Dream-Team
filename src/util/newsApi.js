export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
