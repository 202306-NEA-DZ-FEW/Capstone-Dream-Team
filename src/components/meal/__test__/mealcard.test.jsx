import Mealcard from "../mealcard";
import renderer from "react-test-renderer";
import i18n from "../../../util/i18n";
import { I18nextProvider } from "react-i18next";
import { getAuth } from "@/util/firebase";

// Mock the Firebase services
jest.mock("@/util/firebase", () => ({
    getAuth: jest.fn(),
    initializeApp: jest.fn(), // Mock initializeApp to avoid Firebase initialization
}));

// Mock the AddToCartButton component
jest.mock("../../Cart/addToCartButton", () => {
    return function AddToCartButton() {
        return <div>AddToCartButton component</div>;
    };
});

// Mock the useRouter hook
jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

const price = "400";
const maxMeals = "25";
const name = "Msemen";
const imageUrl = "https://via.placeholder.com/444x530";
const mealDetail = {
    description: "Msemen",
    imageUrl: "https://via.placeholder.com/444x530",
    maxMeals: "25",
    name: "Msemen",
    price: "400",
    quantity: "1",
    restaurantId: "RyOYSd3J67hHqnKWC0dcp7N05Jd2",
};
const mealObject = {
    description: "Msemen",
    imageUrl: "https://via.placeholder.com/444x530",
    maxMeals: "25",
    name: "Msemen",
    price: "400",
    quantity: "1",
    restaurantId: "RyOYSd3J67hHqnKWC0dcp7N05Jd2",
};

it("renders correctly", () => {
    // Mock Firebase services
    getAuth.mockReturnValue();
    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <Mealcard
                    name={name}
                    price={price}
                    maxMeals={maxMeals}
                    imageUrl={imageUrl}
                    mealDetail={mealDetail}
                    mealObject={mealDetail}
                />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
