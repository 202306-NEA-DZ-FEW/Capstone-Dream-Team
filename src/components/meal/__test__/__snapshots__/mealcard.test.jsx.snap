// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mb-10 w-80 h-96 relative"
>
  <div
    className="w-80 h-96 left-0 top-[92px] absolute  rounded-3xl shadow bg-orange-100"
  >
    <div
      className="top-[160px] absolute flex-col left-1/2 -translate-x-1/2 "
    >
      <p
        className="flex text-black text-xl font-bold justify-center items-center font-roboto whitespace-nowrap pb-2"
      >
        Msemen
      </p>
      <div
        className="  flex text-neutral-400 text-lg font-semibold font-['Roboto'] justify-center item-center"
      >
        400
         Â£
      </div>
    </div>
    <div
      className="w-72 h-px left-1/2 -translate-x-1/2 top-[243px] absolute border-2 border-neutral-400 border-opacity-20"
    />
    <div
      className="w-60 h-11 left-1/2 -translate-x-1/2 top-[329px] absolute"
    >
      <div>
        AddToCartButton component
      </div>
    </div>
    <div
      className="mt-[267px] pl-5 pr-5 flex justify-between items-center"
    >
      <div
        className="text-neutral-400 text-base font-medium font-['Roboto']"
      >
        cartPage.card.quantityLeft
        : 
        25
      </div>
      <ul
        className="w-32 h-8 flex items-center"
      >
        <button
          className="w-9 h-8 bg-zinc-100 border border-stone-300 justify-center items-center gap-2.5 flex"
          onClick={[Function]}
        >
          <div
            className="justify-center items-center gap-3.5 flex"
          >
            <div
              className="text-neutral-700 text-lg font-medium font-['Poppins']"
            >
              -
            </div>
          </div>
        </button>
        <input
          className="w-16 h-8 bg-zinc-100 border border-stone-300 justify-center text-center gap-2.5 flex"
          id="input"
          onChange={[Function]}
          type="text"
          value={1}
        />
        <button
          className="w-9 h-8 bg-zinc-100 border border-stone-300 justify-center items-center gap-2.5 flex"
          onClick={[Function]}
        >
          <div
            className="justify-center items-center gap-3.5 flex"
          >
            <div
              className="text-neutral-700 text-lg font-medium font-['Poppins']"
            >
              +
            </div>
          </div>
        </button>
      </ul>
    </div>
  </div>
  <img
    className="w-60 h-60 left-1/2 -translate-x-1/2 top-0 absolute rounded-full"
    src="https://via.placeholder.com/444x530"
  />
</div>
`;
